From: Ladislav Michl <ladis@linux-mips.org>
Date: Mon, 31 May 2021 15:00:21 +0200
Subject: [PATCH] Compilation fails because of kernel header inclusion

Fixes the following error:

In file included from interface.c:27:
/opt/crosstool/gcc-4.1.1-glibc-2.4/arm-unknown-linux-gnu/lib/gcc/arm-unknown-linux-gnu/4.1.1/../../../../arm-unknown-linux-gnu/sys-include/sys/types.h:62: error: conflicting types for 'dev_t'
... etc

Signed-off-by: Ladislav Michl <ladis@linux-mips.org>
---
 src/interface.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/interface.c b/src/interface.c
index e5038f41ba52..352b4799ef4b 100644
--- a/src/interface.c
+++ b/src/interface.c
@@ -23,11 +23,9 @@
 #endif
 
 #include <linux/sockios.h>
-#include <linux/if_ether.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
-#include <linux/if.h>
 #include <syslog.h>
 #include <string.h>
 #include <errno.h>
@@ -43,6 +41,8 @@
 
 #include <libdaemon/dlog.h>
 
+#define ETH_ALEN	6
+
 void interface_up(int fd, char *iface) {
     struct ifreq ifr;
 
